src/fluree/db/transact.cljc

```diff
@@ -205,10 +207,21 @@
 
 (defn publish-commit
   "Publishes commit to all nameservices registered with the ledger."
-  [{:keys [primary-publisher secondary-publishers] :as _ledger} commit-jsonld]
+  [{:keys [primary-publisher secondary-publishers alias] :as _ledger} commit-jsonld]
   (go-try
-    (let [result (<? (nameservice/publish primary-publisher commit-jsonld))]
-      (nameservice/publish-to-all commit-jsonld secondary-publishers)
+    (let [existing-record (when primary-publisher
+                            (<? (nameservice/lookup primary-publisher alias)))
+          branch-metadata (when existing-record
```

Comment:
And the `primary-publisher` should always have an existing record for the ledger, I think `lookup` will throw if it doesn't find anything.